pkg/utils/whisper/whisper.go

```diff
@@ -0,0 +1,79 @@
+package whisper
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	whisper "github.com/rotationalio/whisper/pkg/api/v1"
+)
+
+const (
+	accessDefault int    = 3
+	endpoint      string = "https://api.whisper.rotational.dev"
+)
+
+var (
+	client            whisper.Service
+	initError         error
+	initClient        sync.Once
+	expirationDefault time.Time = time.Now().AddDate(0, 0, 7) // unfortunately, time.Time values cannot be constant in golang
+)
+
+// Creates a Whisper secret and returns the link to the Whisper UI where a user can access the secret value. If the
+// password is specified, then the user will need to enter the password to access the secret. The accesses and
+// expiration date options limit how long the secret is available. By default these values are 3 accesses and 7 days
+// from today. An error is returned if the secret cannot be created with Whisper.
+func CreateSecretLink(secret string, password string, accesses int, expirationDate time.Time) (link string, err error) {
+	// Ensure the secret is not empty
+	if secret == "" {
+		return "", errors.New("a secret is required to generate a Whisper link")
+	}
+
+	// If accesses if not set, set it to the default
+	if accesses <= 0 {
+		accesses = accessDefault
+	}
+
+	// Ensure the expiration date is valid
+	if expirationDate.IsZero() {
+		expirationDate = expirationDefault
+	}
+
+	// Create the whisper client, using sync.Once to ensure we only instantiate
+	// a client once
+	initClient.Do(func() {
+		client, initError = whisper.New(endpoint)
```

Comment:
I think what @bbengfort is suggesting is that if the live test environment variable is not set, we create a test server using `httptest.NewServer` similar to what the whisper tests do [here](https://github.com/rotationalio/whisper/blob/3fdd7139fe19681ee95adf7b45b43863c7f0db4f/pkg/api/v1/client_test.go#L118-L137). That allows us to control the reply, and then if we pass in the URL from the httptest server into the whisper API instead of the real endpoint it will just call the test server.

Reply:
I've utilized httptest.NewServer and split the tests in two, one live, one using the test server. Like Ben recommended, the live test only runs if the GDS_TEST_LIVE_WHISPER environment variable is set.